docs/source/concepts/event.rst

```diff
@@ -9,8 +9,133 @@ The Event System
+including an index into the population table that contains all affected
+simulants, the simulation time at which the event was emitted, the time step
+size at the time of emission, and any additional user-provided data. ``vivarium``
+manages these events by requiring that each type of event be unique
+(e.g., there is only one ``simulation_end`` event and its type (or name - the
+two are equivalent because of the uniqueness) is "simulation_end"). Individual
+components are then allowed to register ``listeners`` (i.e., callable functions
+to respond on the emission of an event) to each event. When an event is emitted,
+all listeners that have registered themselves to that event type are called.
+
+Lifecycle Events
+----------------
+
+The simulation engine itself is responsible for the emission of a dedicated set
+of events that determine the progression of time in the simulation. Each time
+step in the simulation corresponds to the emission of a set of ``time_step``
+events. :ref:`Components <components_concept>` can register themselves as
+listeners to these events and thus take action on each time step or other
+key simulation phase.
+
+The following table depicts the events emitted by the simulation engine itself
+and the lifecycle phase when they are emitted. See the
+:ref:`lifecycle concept note <lifecycle_concept>` for more information about
+these phases.
+
+.. list-table:: **Events Emitted by the Simulation Engine**
+    :header-rows: 1
+    :widths: 30, 40
+
+    *   - Lifecycle Phase
+        - Events
+    *   - | Post-setup
+        - | ``post_setup``
+    *   - | Main Event Loop
+        - | ``time_step_prepare``
+          | ``time_step``
+          | ``time_step__cleanup``
+          | ``collect_metrics``
+    *   - | Finalization
+        - | ``simulation_end``
+
+
+Interacting with Events
+-----------------------
+
+The :class:`EventInterface <vivarium.framework.event.EventInterface>` is available
+off the :ref:`Builder <builder_concept>` and provides two options for interacting
+with the event system:
+
+1. :func:`register_listener <EventInterface.register_listener()>` to add a
+listener to a given event to be called on emission
+
+2. :func:`get_emitter <vivarium.framework.event.EventInterface.get_emitter()>` to
+retrieve a callable emitter for a given event
+
+Although methods for both getting emitters and registering listeners are provided,
+it is strongly encouraged that only the registering listeners aspect is used.
+
+
+Registering Listeners
++++++++++++++++++++++
+In order to register a listener to an event to respond when that event is
+emitted, we can use the :func:`register_listener <EventInterface.register_listener()>`.
+The listener itself should be a callable function that takes as its only argument
+the :class:`Event <vivarium.framework.event.Event>` that is emitted.
+
+Suppose we wish to track how many simulants are affected each time step. We
```

Comment:
this part felt a bit more like a tutorial but it felt odd to talk about registering listeners and not explain how to do it